src/descriptor/mod.rs

```diff
@@ -234,13 +233,68 @@ pub trait PkTranslate<P: MiniscriptKey, Q: MiniscriptKey> {
     where
         Fpk: FnMut(&P) -> Q,
         Fpkh: FnMut(&P::Hash) -> Q::Hash,
-        Q: MiniscriptKey,
     {
         self.translate_pk::<_, _, ()>(|pk| Ok(translatefpk(pk)), |pkh| Ok(translatefpkh(pkh)))
             .expect("infallible translation function")
     }
 }
 
+/// Variant of `PkTranslate` where P and Q both have the same hash
+/// type, and the hashes can be converted by just cloning them
+pub trait PkTranslate1<P: MiniscriptKey, Q: MiniscriptKey<Hash = P::Hash>>:
+    PkTranslate<P, Q>
+{
+    /// Translate a struct from one generic to another where the
+    /// translation for Pk is provided by translatefpk
+    fn translate_pk1<Fpk, E>(
+        &self,
+        translatefpk: Fpk,
+    ) -> Result<<Self as PkTranslate<P, Q>>::Output, E>
+    where
+        Fpk: FnMut(&P) -> Result<Q, E>,
+    {
+        self.translate_pk(translatefpk, |h| Ok(h.clone()))
+    }
+
+    /// Translate a struct from one generic to another where the
+    /// translation for Pk is provided by translatefpk
+    fn translate_pk1_infallible<Fpk: FnMut(&P) -> Q>(
+        &self,
+        translatefpk: Fpk,
+    ) -> <Self as PkTranslate<P, Q>>::Output {
+        self.translate_pk_infallible(translatefpk, P::Hash::clone)
+    }
+}
+impl<P: MiniscriptKey, Q: MiniscriptKey<Hash = P::Hash>, T: PkTranslate<P, Q>> PkTranslate1<P, Q>
+    for T
+{
+}
+
+/// Variant of `PkTranslate` where P's hash is P, so the hashes
+/// can be converted by reusing the key-conversion function
+pub trait PkTranslate2<P: MiniscriptKey<Hash = P>, Q: MiniscriptKey>: PkTranslate<P, Q> {
```

Comment:
I know you probably thought of this :P , but should we give these full names? I think everything we write translate downstream, we will wonder what 1 or 2 stands for.

Reply:
My reasoning was roughly:
* There is readability value in having short names, that you have to trade off against the readability of being explicit
* Usually, the specific `translate_pk` method being called is irrelevant or clear from context